File: lib/workflow_cancel_parallel/workflow_cancel_parallel_activity.ex
defmodule WorkflowCancelParallel.Activity do
  use TemporalSdk.Activity
  @moduledoc false

  @impl true
  def execute(_context, [activities]) when is_list(activities) do
    IO.puts("Running cleanup activity. Canceled activities count: #{length(activities)}.")
    ["ok"]
  end
end

File: src/workflow_cancel_parallel/workflow_cancel_parallel_activity.erl
-module(workflow_cancel_parallel_activity).

% elp:ignore W0012 W0040
-moduledoc false.

-export([execute/2]).

-include_lib("temporal_sdk/include/activity.hrl").

execute(_Context, [Activities]) when is_list(Activities) ->
    io:fwrite("Running cleanup activity. Canceled activities count: ~b.~n", [length(Activities)]),
    [~"ok"].
